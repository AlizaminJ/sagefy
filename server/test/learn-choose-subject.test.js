const Joi = require('joi')
const request = require('supertest')
const { app } = require('../index')
const {
  GQL: { learnChooseSubject },
  getLoginToken,
  getData,
} = require('./_util')

describe('learn-choose-subject', () => {
  it.skip('Requires a parent subject_id', async () => {
    expect(true).toBe(false)
  })

  it('Should not choose subjects with unmet befores', async () => {
    const token = await getLoginToken('Doris')
    const data = await getData()
    return request(app)
      .post('/graphql')
      .send({
        query: learnChooseSubject,
        variables: {
          subjectId: data.subjects.all.entity_id,
        },
      })
      .set('Authorization', `Bearer ${token}`)
      .expect(({ body }) => {
        Joi.assert(
          body,
          Joi.object({
            data: Joi.object({
              selectSubjectToLearn: Joi.object({
                nodes: Joi.array()
                  .items(
                    Joi.object({
                      entityId: Joi.string()
                        .guid()
                        .required(),
                      name: Joi.string().required(),
                      body: Joi.string().required(),
                    })
                  )
                  .required(),
              }),
            }),
          })
        )
        expect(
          body.data.selectSubjectToLearn.nodes.map(({ name }) => name)
        ).toMatchSnapshot()
      })
  })

  it.skip('Should not choose subjects with no direct cards', async () => {
    expect(true).toBe(false)
  })

  it.skip('Should order subjects by depth of afters', async () => {
    expect(true).toBe(false)
  })

  it.skip('Should limit to 5 options', async () => {
    expect(true).toBe(false)
  })

  it.skip('Should do nothing if complete', async () => {
    expect(true).toBe(false)
  })

  it.skip('Can choose itself if direct cards and no child subjects', () => {
    expect(true).toBe(false)
  })
})
