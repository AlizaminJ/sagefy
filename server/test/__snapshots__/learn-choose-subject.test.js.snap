// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`learn-choose-subject Should not choose subjects with unmet befores 1`] = `
Array [
  "Oscillators",
  "Hearing Curves",
  "Spatialization",
  "Modulation",
  "Delay",
]
`;
